docs/design/task-DAG-scheduling.md

```diff
@@ -0,0 +1,85 @@
+# Task-level DAG scheduling policy
+
+@[hwdef](https://github.com/hwdef); July 19th, 2021
+
+## Introduction
+
+This feature provides the ability to customize the order in which tasks are launched.
+
+## Scenarios
+
+* mpi job. the master needs to wait for the worker to start before starting, If the master is already started, but the worker is not yet started, the master will restart, which will add unnecessary waste of resources ,in this case, mpiworker needs to be in the running state, and then create the pod for mpimaster..
+
+* Suppose there are two tasks and task 2 needs to use the calculation results of task 1,in this case, task1 needs to be in the complete state, and then create the pod for task2.
+
+* some ETL applications
+
+## Requirement
+
+Each task can specify which one or more tasks it depends on, and it can specify the time point or interface for triggering.
+
+Examples of time points:
+
+* running
+* has been running for 10s
+* complete
+
+Examples of interfacets:
+
+* Check if a file named raw exists in the /data directory.
+* Check if the specified port is open. such as 2222.
+* Check if the http GET request to the specified path was successfully accessed.such as /health.
+
+## Design
+
+1. Add a field that is an array named `startPolicy` in task spec
+
+```go
+type TaskSpec struct {
+    ...
+	StartPolicy []StartPolicySpec
+}
+```
+2. It contains three fields, `dependOn`, `trigger`, `probe`
+
+```go
+type StartPolicySpec struct {
+	DependOn string
+	Trigger  TriggerSpec
+	Probe    *v1.Probe
+}
+```
+2.1 `dependOn` is a string indicating the name of the task that the current task depends on
+
+2.2 `trigger` indicates the point in time when the trigger, it has two fields, `status` and `time`, `status` indicates the status of the task, `time` indicates the duration of time since the state was reached,it can be empty, indicating 0 seconds.
+
+```go
+type TriggerSpec struct {
+	Status string
+	Time   time.Duration
```

Comment:
Does `Time` means the time period after reaching the state or the period after the task starting to run

Reply:
`time` is the time period after reaching the state

Reply:
Such as the task has been running for 10s

Reply:
ls there any usage scenarios? Generally, l think `Status` is enough, wdyt?

Reply:
I haven't thought of any scenarios yet, but I think it will be more flexible with the addition of the time field.

Reply:
In terms of realization, if we want to support this, we also need to record the transition time of task (e.g maybe the time when all replicas under one task converting to running), which is not available right now

Reply:
yes, you are right, But I think maybe we can just use task to meet the minavailable time

Reply:
> yes, you are right, But I think maybe we can just use task to meet the minavailable time

That's an available solution.